File: app/src/main/java/sabalan/paydar/mohtasham/ezibazi/controller/api_service/payment/FinanceService.java
package sabalan.paydar.mohtasham.ezibazi.controller.api_service.payment;

import android.content.Context;

import org.json.JSONException;
import org.json.JSONObject;



import sabalan.paydar.mohtasham.ezibazi.controller.api_service.ApiRequest;
import sabalan.paydar.mohtasham.ezibazi.controller.api_service.Urls;

public class FinanceService {
  private Context context;
  public FinanceService(Context context){
    this.context = context;
  }


  public void getPayUrl(JSONObject jsonObject, final onCreditComplete onCreditComplete){
    ApiRequest apiRequest = new ApiRequest(context);
    apiRequest.request(ApiRequest.POST, Urls.INCREASE_CREDIT, jsonObject, true, new ApiRequest.onDataReceived() {
      @Override
      public void onReceived(JSONObject response, int status, String message, boolean error) {
        if (error){
          onCreditComplete.onComplete(status, message, "");
          return;
        }

        String url = "";
        try {
          url = response.getString("data");
          onCreditComplete.onComplete(status, message, url);
        } catch (JSONException e) {
          e.printStackTrace();
        }

      }
    });

  }










  public interface onCreditComplete{
    void onComplete(int status, String message, String url);
  }



}

File: app/src/main/java/sabalan/paydar/mohtasham/ezibazi/api_service/payment/FinanceService.kt
package sabalan.paydar.mohtasham.ezibazi.api_service.payment

import android.content.Context

import org.json.JSONException
import org.json.JSONObject


import sabalan.paydar.mohtasham.ezibazi.api_service.ApiRequest
import sabalan.paydar.mohtasham.ezibazi.api_service.Urls

class FinanceService(private val context: Context) {


    fun getPayUrl(jsonObject: JSONObject, onCreditComplete: onCreditComplete) {
        val apiRequest = ApiRequest(context)
        apiRequest.request(ApiRequest.POST, Urls.INCREASE_CREDIT, jsonObject, true, object : ApiRequest.onDataReceived {
            override fun onReceived(response: JSONObject?, status: Int, message: String, error: Boolean) {
                if (error) {
                    onCreditComplete.onComplete(status, message, "")
                    return
                }

                var url = ""
                try {
                    url = response!!.getString("data")
                    onCreditComplete.onComplete(status, message, url)
                } catch (e: JSONException) {
                    e.printStackTrace()
                }

            }
        })

    }


    interface onCreditComplete {
        fun onComplete(status: Int, message: String, url: String)
    }


}
